modules/javafx.controls/src/main/java/javafx/scene/control/skin/ToolBarSkin.java

```diff
@@ -778,11 +767,64 @@ private void fire() {
                 popup.hide();
             } else {
                 popup.getItems().clear();
-                popup.getItems().addAll(menuItems);
+                popup.getItems().addAll(createMenuItems());
                 popup.show(downArrow, Side.BOTTOM, 0, 0);
             }
         }
 
+        private List<MenuItem> createMenuItems() {
+            List<MenuItem> menuItems = new ArrayList<>();
+            for (Node node : overflowItems) {
+                if (node instanceof Separator) {
+                    menuItems.add(new SeparatorMenuItem());
+                } else {
+                    CustomMenuItem customMenuItem = new CustomMenuItem(node);
+
+                    // RT-36455:
```

Comment:
since we moved this code, could we change this line to
`     // RT-36455 (JDK-8096292):`

and also make sure that this change introduced no regression w.r.t. JDK-8096292 ?
(My limited testing indicates that it still works as expected, but please double check)

Reply:
I tested it with the Modena application. The bug is not reproducible.

Reply:
I changed the comment. See [e63b0b9](https://github.com/openjdk/jfx/pull/1434/commits/e63b0b9b313131460fbe154ce8af164a3a97672f)